forc-plugins/forc-client/tests/deploy.rs

```diff
@@ -405,3 +410,157 @@ fn test_deploy_interactive_wrong_password() -> Result<(), rexpect::error::Error>
     node.kill().unwrap();
     Ok(())
 }
+
+#[tokio::test]
+async fn chunked_deploy() {
+    let (mut node, port) = run_node();
+    let tmp_dir = tempdir().unwrap();
+    let project_dir = test_data_path().join("big_contract");
+    copy_dir(&project_dir, tmp_dir.path()).unwrap();
+    patch_manifest_file_with_path_std(tmp_dir.path()).unwrap();
+
+    let pkg = Pkg {
+        path: Some(tmp_dir.path().display().to_string()),
+        ..Default::default()
+    };
+
+    let node_url = format!("http://127.0.0.1:{}/v1/graphql", port);
+    let target = NodeTarget {
+        node_url: Some(node_url),
+        target: None,
+        testnet: false,
+    };
+    let cmd = cmd::Deploy {
+        pkg,
+        salt: Some(vec![format!("{}", Salt::default())]),
+        node: target,
+        default_signer: true,
+        ..Default::default()
+    };
+    let deployed_contract = deploy(cmd).await.unwrap().remove(0);
+    node.kill().unwrap();
+
+    let num_chunks = deployed_contract.chunks.len();
+    assert_eq!(num_chunks, 4);
+}
+
+#[tokio::test]
+async fn chunked_deploy_re_routes_call() {
+    let (mut node, port) = run_node();
+    let tmp_dir = tempdir().unwrap();
+    let project_dir = test_data_path().join("big_contract");
+    copy_dir(&project_dir, tmp_dir.path()).unwrap();
+    patch_manifest_file_with_path_std(tmp_dir.path()).unwrap();
+
+    let pkg = Pkg {
+        path: Some(tmp_dir.path().display().to_string()),
+        ..Default::default()
+    };
```

Comment:
This piece of code is repeated in almost every test, and should be refactored into a helper function like:

```rust
fn create_tmp_pkg(project_name: &str) -> Pkg {
    let tmp_dir = tempdir().expect("temp dir");
    let project_dir = test_data_path().join(project_name);
    copy_dir(&project_dir, tmp_dir.path()).expect("copy dir");
    patch_manifest_file_with_path_std(tmp_dir.path()).expect("patch manifest");

    Pkg {
        path: Some(tmp_dir.path().display().to_string()),
        ..Default::default()
    }
}
```

Reply:
Added this to our clean-up list #6259 to get this in asap, I will clean bunch of stuff, after we get through this release